src/noc/containers/noc-field/utils/noc-error.pipe.ts

```diff
@@ -0,0 +1,49 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { ValidationErrors } from '@angular/forms';
+
+@Pipe({
+  name: 'nocError'
+})
+export class NocErrorPipe implements PipeTransform {
+
+  public transform(value: ValidationErrors): string {
+
+    if (!value) {
+      return '';
+    }
+    const keys = Object.keys(value);
+
+    if (!keys.length) {
+      return '';
+    }
+    if (keys[0] ===  'required') {
```

Comment:
Should be a switch statement.

Reply:
Agreed, but I would keep the same if/else block to align with ccd toolkit for further maintainence.

Reply:
Is this identical code to what's in the CCD toolkit? If so, I don't understand why it's duplicated; if not, I don't understand how it helps with maintenance...?

Reply:
Those in ccd is called CASE field type, in NoC they are not case field type but question field type, that's the reason we can't reuse the ccd toolkit ones, also there are some requirement differences on toolkit which can'be be reused in NoC.